src/auth.py

```diff
@@ -0,0 +1,929 @@
+# -*- coding: utf-8 -*-
+# Copyright 2022 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Base authentication analyzer"""
```

Comment:
The filename "auth.py" is confusing in the context of an OpenRelik Log Analyser Worker which can contain different type of analyzers. This class essentially analysis auth log records for different types of anomalies. Maybe "auth_log_analyzers" is more appropriate. wdyt?